lib/charms/certificate_transfer_interface/v1/certificate_transfer.py

```diff
@@ -0,0 +1,381 @@
+                self.relationship_name,
+            )
+            return
+
+        for relation in relations:
+            existing_data = self._get_relation_data(relation)
+            existing_data.add(certificate)
+            self._set_relation_data(relation, existing_data)
+
+    def remove_certificate(
+        self,
+        certificate: str,
+        relation_id: Optional[int] = None,
+    ) -> None:
+        """Remove a given certificate from relation data.
+
+        Removes certificate from all relations if relation_id not given
+
+        Args:
+            certificate (str): Certificate
+            relation_id (int): Relation ID
+
+        Returns:
+            None
+        """
+        relations = self._get_relevant_relations(relation_id)
+        if len(relations) == 0:
+            logger.error(
+                "At least 1 matching relation ID not found with the relation name '%s'",
+                self.relationship_name,
+            )
+            return
+
+        for relation in relations:
+            existing_data = self._get_relation_data(relation)
+            existing_data.discard(certificate)
+            self._set_relation_data(relation, existing_data)
+
+    def _set_relation_data(self, relation: Relation, data: Set[str]) -> None:
+        """Set the given relation data."""
+        relation.data[self.model.app]["certificates"] = json.dumps(list(data))
+
+    def _get_relation_data(self, relation: Relation) -> Set[str]:
+        """Get the given relation data."""
+        data = relation.data[self.model.app].get("certificates", "[]")
+        try:
+            loaded_data = json.loads(data)
+            if type(loaded_data) is not list:
+                raise TypeError
+            cleaned_data = set(json.loads(data))
+        except (json.decoder.JSONDecodeError, TypeError):
+            logger.error(
+                (
+                    "Error parsing relation databag. "
+                    "Make sure not to interact with the databags "
+                    "except using the public methods in the provider library "
+                    "and use version V1."
+                )
+            )
+            return set()
+        return cleaned_data
+
+    def _get_relevant_relations(self, relation_id: Optional[int] = None) -> List[Relation]:
+        """Get the relevant relation if relation_id is given, all relations otherwise."""
+        if (
+            relation_id is not None
+            and (
+                relation := self.model.get_relation(
+                    relation_name=self.relationship_name, relation_id=relation_id
+                )
+            )
+            and relation.active
+        ):
+            return [relation]
+
+        return list(self.model.relations[self.relationship_name])
+
+
+class CertificateSetUpdatedEvent(EventBase):
+    """Charm Event triggered when a TLS certificate is available."""
```

Comment:
How about something like?
```suggestion
    """Charm Event triggered when a CA certificates list is updated."""
```

Reply:
changed them to be more accurate. I don't want to specify that it's a CA certificate, since this relation could be used to send any type of certificate to be trusted.